libqtile/widget/systray.py

```diff
@@ -47,8 +47,17 @@ class Icon(window._Window):
     def __init__(self, win, qtile, systray):
         window._Window.__init__(self, win, qtile)
         self.systray = systray
+        # win.get_name() may return None when apps provide a temporary window before the icon window
+        # we need something in self.name in order to sort icons so we use the window's WID.
+        self.name = win.get_name() or str(win.wid)
```

Comment:
Is it a 'temporary window', or have they just not set any appropriate hints yet?

Reply:
It has a different WID so I'm assuming temporary.

Reply:
Hmm, I wonder what they are up to...

Reply:
Not sure but, when I launch `blueman-applet`, I first get a window with the class `blueman-tray` then it's destroyed and replaced by one with a `blueman` class. Who knows what's going on but they are different windows.